### PR TITLE
refactor(core/jquery-enhanced): move getDfnTitles to vanilla js

### PR DESCRIPTION
**Pull request by sidvishnoi**
As a part of https://github.com/w3c/respec/issues/1699.

Do we still need to keep the backward compatibility for `dfn-type => data-dfn-type`, `lt => data-lt` etc? 

Presently just moved code and removed jquery. Will try to refactor this function more next.

### CODE REVIEW

**Comment by saschanaz** [comment]
>Do we still need to keep the backward compatibility for dfn-type => data-dfn-type, lt => data-lt etc?

I think we already removed it but not sure...

**Comment by marcoscaceres** [comment]
> Do we still need to keep the backward compatibility for dfn-type => data-dfn-type, lt => data-lt etc?

As @saschanaz, I think we removed that. But if there are some strays, it's ok to remove them.

File: src/core/utils.js [hunk 1]
```diff
@@ -626,3 +626,60 @@ export function getTextNodes(el, exclusions = []) {
...
+ *   <dfn> element (but can apply to other as well).
+ * if args.isDefinition is true, then the element is a definition, not a
+ *   reference to a definition. Any @title or @lt will be replaced with
+ *   @data-lt to be consistent with Bikeshed / Shepherd.
+ * This method now *prefers* the data-lt attribute for the list of
+ *   titles. That attribute is added by this method to dfn elements, so
+ *   subsequent calls to this method will return the data-lt based list.
+ * @param {Element} elem
+ * @param {Object} args
+ * @returns {Array:String} array of title strings
```
Comment (saschanaz) [inline src/core/utils.js]: Maybe just `string[]` according to the docs: http://usejsdoc.org/tags-type.html

File: src/core/utils.js [hunk 2]
```diff
@@ -626,3 +626,60 @@ export function getTextNodes(el, exclusions = []) {
...
+  if (elem.dataset.lt) {
+    // prefer @data-lt for the list of title aliases
+    titleString = elem.dataset.lt.toLowerCase();
+    if (normText !== "" && !titleString.startsWith(`${normText}|`)) {
+      // Use the definition itself, so to avoid having to declare the definition twice.
+      titleString = titleString + "|" + normText;
+    }
+  } else if (
+    elem.childNodes.length === 1 &&
+    [...elem.children].filter(el => el.matches("abbr, acronym")).length === 1 &&
```
Comment (saschanaz) [inline src/core/utils.js]: I would recommend `elem.querySelectorAll` or even `elem.getElementsByTagName`.

`acronym` is an obsolete element in HTML5, do we still need it? @marcoscaceres

File: src/core/utils.js [hunk 3]
```diff
@@ -626,3 +626,60 @@ export function getTextNodes(el, exclusions = []) {
...
+    elem.querySelector(":first-child").getAttribute("title")
```
Comment (saschanaz) [inline src/core/utils.js]: `elem.children[0]`?

File: src/core/utils.js [hunk 4]
```diff
@@ -626,3 +626,60 @@ export function getTextNodes(el, exclusions = []) {
...
+  ) {
+    titleString = elem.querySelector(":first-child").getAttribute("title");
```
Comment (saschanaz) [inline src/core/utils.js]: Same here.

File: src/core/utils.js [hunk 5]
```diff
@@ -626,3 +626,60 @@ export function getTextNodes(el, exclusions = []) {
...
+  } else {
+    titleString =
+      elem.textContent === '""' ? "the-empty-string" : elem.textContent;
```
Comment (sidvishnoi) [inline src/core/utils.js]: @saschanaz I don't understand this `'""'` test. Isn't that simply testing empty string? Or are there cases when `textContent === '""'`

Comment (marcoscaceres) [reply src/core/utils.js]: The intention was to test if it's exactly what's there (i.e., a pair of quotes).

Comment (marcoscaceres) [reply src/core/utils.js]: screenshot 2018-07-10 08 13 28

Comment (sidvishnoi) [reply src/core/utils.js]: So, one may write `<a>""</a>`?

Comment (marcoscaceres) [reply src/core/utils.js]: Yep.

File: src/core/utils.js (hunk 2 above)
Comment (sidvishnoi) [reply src/core/utils.js]: I found some instances of `acronym` in some `respecConfig`s and tests/specifcations in ReSpec. Not sure if we should remove its support. WDYT @marcoscaceres?

**Comment by marcoscaceres** [comment]
Re: acronym, please kill it if you haven't already!

File: src/core/jquery-enhanced.js [hunk 6]
```diff
@@ -50,57 +50,7 @@ window.$.fn.renameElement = function(name) {
 // This method will publish a warning if a title is used on a definition
 // instead of an @lt (as per specprod mailing list discussion).
 window.$.fn.getDfnTitles = function(args) {
```
Comment (saschanaz) [inline src/core/jquery-enhanced.js]: @sidvishnoi @marcoscaceres Why do we have to keep this extension? Is there any compat issue?

Comment (saschanaz) [reply src/core/jquery-enhanced.js]: Removing this will require fixing tests, though.

Comment (saschanaz) [reply src/core/jquery-enhanced.js]: I think we should mark them as deprecated, as it's easy enough in this case.

Comment (marcoscaceres) [reply src/core/jquery-enhanced.js]: Absolutely. We should console.warn and start seeing if there will be any damage in yanking support.

**Pull request merged**